modules/videoio/test/test_gstreamer.cpp

```diff
@@ -0,0 +1,259 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+#include "opencv2/videoio/videoio_c.h"
+#ifdef HAVE_GSTREAMER
+
+namespace opencv_test
+{
+typedef cv::Point_<double> Point2d;
+typedef tuple<string, string> Param;
+
+class Videoio_Gstreamer_Test : public testing::TestWithParam<Param>
+{
+protected:
+    Videoio_Gstreamer_Test()
+    {
+        format = get<0>(GetParam());
```

Comment:
It is better to use more parameters to simplify test behavior:
```.cpp
// parameters struct
struct TheP {
    string gstFormat;
    Size frameSize;
    Size matSize;
    bool isDecodeNeeded;
    int conversionToBGR;
};
// helper for console printing
std::ostream & operator<<(std::ostream & out, const TheP & p)
{
    out << "{" << p.gstFormat << "; " << p.frameSize << "}";
    return out;
}
// configurations
TheP vals[] = {
    {"video/x-raw,format=BGR", Size(640, 480), Size(640, 480), false, -1},
    {"video/x-raw,format=NV12", Size(640, 480), Size(640, 720), false, COLOR_YUV2BGR_NV12},
    //...
};
// instantiation
INSTANTIATE_TEST_CASE_P(videoio, Videoio_Gstreamer_Test, testing::ValuesIn(vals));
```
Using this way you will be able to omit complex conditions like `if (format == "video/x-bayer") ... else if (...) ...`